backend/src/test/java/com/github/johannesthorbergsson/backend/orders/OrderControllerTest.java

```diff
@@ -100,4 +99,96 @@ void addOrder_whenOrderRequest_thenReturnOrder() throws Exception {
                 .andExpect(jsonPath("$.id").isNotEmpty());
 
     }
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "steven")
+    void updateOrder_whenValidRequest_thenReturnUpdatedOrder() throws Exception {
+        orderRepository.save(testOrder);
+        mockMvc.perform(put("/api/orders/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("""
+                          {
+                              "bikeId": "bid",
+                              "description": "New Tyre",
+                              "workshop": "Workshop42",
+                              "status": "OPEN",
+                              "componentsToReplace": [
+                                  {
+                                      "category": "Tyre",
+                                      "type": "Pirelli",
+                                      "age": 1337
+                                  }
+                              ]
+                         }
+                    """)
+                .with(csrf()))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                          {
+                              "id": "1",
+                              "bikeId": "bid",
+                              "description": "New Tyre",
+                              "workshop": "Workshop42",
+                              "username": "steven",
+                              "status": "OPEN",
+                              "componentsToReplace": [
+                                  {
+                                      "category": "Tyre",
+                                      "type": "Pirelli",
+                                      "age": 1337
+                                  }
+                              ]
+                         }
+                    """));
+    }
+    @Test
+    @DirtiesContext
+    @WithMockUser
+    void updateOrder_whenOrderNotFound_thenThrowNoSuchOrderException() throws Exception {
+        mockMvc.perform(put("/api/orders/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("""
+                    {
+                          "bikeId": "bid",
```

Comment:
This could simplify the code and convert your java class to json:
`
ObjectMapper mapper = new ObjectMapper();
String jsonObj = mapper.writeValueAsString(demoTopicJavaDTO);`

Reply:
Amazing.
If I had known this earlier it would have saved me so much test debugging.